provisioner/terraform/parse.go

```diff
@@ -28,6 +34,129 @@ func (s *server) Parse(sess *provisionersdk.Session, _ *proto.ParseRequest, _ <-
 		return provisionersdk.ParseErrorf("load module: %s", formatDiagnostics(sess.WorkDirectory, diags))
 	}
 
+	workspaceTags, err := s.loadWorkspaceTags(ctx, module)
+	if err != nil {
+		return provisionersdk.ParseErrorf("can't load workspace tags: %v", err)
+	}
+
+	templateVariables, err := loadTerraformVariables(module)
+	if err != nil {
+		return provisionersdk.ParseErrorf("can't load template variables: %v", err)
+	}
+
+	return &proto.ParseComplete{
+		TemplateVariables: templateVariables,
+		WorkspaceTags:     workspaceTags,
+	}
+}
+
+var rootTemplateSchema = &hcl.BodySchema{
+	Blocks: []hcl.BlockHeaderSchema{
+		{
+			Type:       "data",
+			LabelNames: []string{"type", "name"},
+		},
+	},
+}
+
+var coderWorkspaceTagsSchema = &hcl.BodySchema{
+	Attributes: []hcl.AttributeSchema{
+		{
+			Name: "tags",
+		},
+	},
+}
+
+func (s *server) loadWorkspaceTags(ctx context.Context, module *tfconfig.Module) (map[string]string, error) {
```

Comment:
This is in a hot-enough path that it may warrant some internal_tests to validate the failure cases.

Reply:
I'd rather not experiment with `tf.Module` as it means that we have to produce an artificial instance of it. In `parse_test.go` we have tests that operate on real .tf files and check the result/output. If it isn't a game changer, I would leave it as is.

WDYT?

Reply:
Done 👍